packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue

```diff
@@ -0,0 +1,410 @@
+<template>
+    <div
+        ref="cookieBanner"
+        :class="[$style['c-cookieBanner'], $style['c-cookieBanner-overlay'], { [$style['c-cookieBanner--hidden']]: hideBanner }]"
+        attr="data-cookie-consent-overlay"
+        :aria-hidden="isHidden">
+        <div
+            v-if="!legacyBanner"
+            :class="$style['c-cookieBanner-card']">
+            <div
+                id="dialog1"
+                :class="$style['c-cookieBanner-content']"
+                role="dialog"
+                aria-labelledby="cookieConsentLabel"
+                aria-modal="true">
+                <h2
+                    id="cookieConsentLabel"
+                    :class="$style['c-cookieBanner-title']">
+                    <a
+                        ref="cookieBannerHeadingFocus"
+                        name="cookieConsentTitle"
+                        tabindex="0"
+                        data-consent-title>
+                        {{ copy.mainTitle }}
+                    </a>
+                </h2>
+
+                <p :class="$style['c-cookieBanner-text']">
+                    {{ copy.text1 }}
+                </p>
+                <p :class="$style['c-cookieBanner-text']">
+                    {{ copy.text2 }}
+                </p>
+                <p
+                    :class="$style['c-cookieBanner-text']">
+                    {{ copy.text3 }}
+                    <a
+                        :href="copy.linkHref"
+                        :class="$style['c-cookieBanner-link']"
+                        target="_blank">
+                        {{ copy.linkText }}</a>{{ copy.text4 }}
+                </p>
+            </div>
+
+            <div :class="$style['c-cookieBanner-CTA']">
+                <button-component
+                    tabindex="0"
+                    role="button"
+                    data-button-accept
+                    is-full-width
+                    @click.native="acceptActions">
+                    {{ copy.buttonText2 }}
+                </button-component>
+                <button-component
+                    tabindex="0"
+                    role="button"
+                    button-type="ghost"
+                    data-button-nonaccept
```

Comment:
I'm not sure about this either. I see you're following this pattern but I'm struggling to understand it. Did you mean to add `data-test-id` here?

Reply:
More leftovers from the GTM code - removed